ugvc/utils/metrics_utils.py

```diff
@@ -331,3 +331,54 @@ def merge_trimmer_histograms(trimmer_histograms: list[str], output_path: str):
     )
     df_merged.to_csv(output_filename, index=False)
     return output_filename
+
+def read_trimmer_failure_codes(trimmer_failure_codes_csv: str, add_total: bool = False) -> pd.DataFrame:
+    """
+    Read a trimmer failure codes csv file
+
+    Parameters
+    ----------
+    trimmer_failure_codes_csv : str
+        path to a Trimmer failure codes file
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with trimmer failure codes
+
+    Raises
+    ------
+    AssertionError
+        If the columns are not as expected
+    """
+    df_trimmer_failure_codes = pd.read_csv(trimmer_failure_codes_csv)
+    expected_columns = [
+        "read group",
+        "code",
+        "format",
+        "segment",
+        "reason",
+        "failed read count",
+        "total read count",
+    ]
+    assert (
```

Comment:
it is better not to use assert in production code (mainly in tests).
A better alternative is to use exceptions.
Please refer to some info on assert usage: https://snyk.io/blog/the-dangers-of-assert-in-python/